web-local/tests/utils/projectPaths.ts

```diff
@@ -0,0 +1,67 @@
+import { asyncWaitUntil } from "@rilldata/web-common/lib/waitUtils";
+import type { V1ListResourcesResponse } from "@rilldata/web-common/runtime-client";
+import { rmSync } from "fs";
+import { join } from "node:path";
+import { cpSync } from "node:fs";
+import treeKill from "tree-kill";
+import { spawnAndMatch } from "web-common/tests/utils/spawn";
+import { getOpenPort } from "web-local/tests/utils/getOpenPort";
+
+export const BASE_PROJECT_DIRECTORY = "temp/test-project";
+export const TEST_PROJECTS = "tests/data/projects";
+
+/**
+ * Prep a project by pre-ingesting and building the tmp folder.
+ */
+export async function prepProject(name: string) {
+  const TEST_PROJECT_SRC_DIRECTORY = join(TEST_PROJECTS, name);
+  const TEST_PROJECT_DIRECTORY = join(BASE_PROJECT_DIRECTORY, name);
+  const TEST_PORT = await getOpenPort();
+  const TEST_PORT_GRPC = await getOpenPort();
+
+  rmSync(TEST_PROJECT_DIRECTORY, { force: true, recursive: true });
+  cpSync(TEST_PROJECT_SRC_DIRECTORY, TEST_PROJECT_DIRECTORY, {
+    recursive: true,
+    force: true,
+  });
+
+  const { process } = await spawnAndMatch(
+    "../rill",
+    [
+      "start",
+      "--no-open",
+      "--port",
+      "" + TEST_PORT,
+      "--port-grpc",
+      "" + TEST_PORT_GRPC,
+      TEST_PROJECT_DIRECTORY,
+    ],
+    new RegExp("Serving Rill on: http://localhost"),
+  );
+
+  await asyncWaitUntil(
```

Comment:
I'd prefer that we use Playwright's built-in `await expect.poll(asyncFn, options)` ([docs](https://playwright.dev/docs/test-assertions#expectpoll))